src/lib/core/SymbolSolver.cpp

```diff
@@ -473,18 +481,29 @@ void SymbolSolver::solveNames(LinuxProcMapEntry * procMapEntry)
 	bool hasEntries = false;
 	const std::string elfFile = procMapEntry->file;
 	std::stringstream addr2lineCmd;
-	addr2lineCmd << "addr2line -C -f -e " << elfFile;
+	const char *prefix = "addr2line -C -f -e ";
+	addr2lineCmd << prefix << elfFile;
+	const size_t prefixLen = addr2lineCmd.str().length();
 	std::vector<CallSite*> lst;
-	
+	std::vector <std::string> theCommands;
+
 	//create addr2line args
 	bool firstNeedAslrScan = true;
 	for (CallSiteMap::iterator it = callSiteMap.begin() ; it != callSiteMap.end() ; ++it)
 	{
 		if (it->second.mapEntry == procMapEntry)
 		{
+			if (addr2lineCmd.str().length() > MAX_ARG_STRLEN-32) {
```

Comment:
Not exactly sure to understand why `-32` ?

I would probably put `prefix+elfFile` in a var and use its size if it is what 32 is related to ?

Reply:
Actually I guess I should change this so that it is adding the length of the next argument.  commit to follow.